### ISSUE TITLE
META_EXIFTOOL MAC time error

### ISSUE THREAD

**Issue by jh1113** [lmco/laikaboss#81]
hi
I have a question about exiftool at laika.py.

When scanning a file, the MAC in exiftool displays the current time.

![image](https://user-images.githubusercontent.com/62418986/131091895-ab59e090-1bd9-42fd-9f4c-117958aeb2ad.png)

![image](https://user-images.githubusercontent.com/62418986/131091811-f2aaa707-f113-428b-a6f7-6516046aca00.png)

Do you know how to solve this?

thank you

**Comment by agrajag9** [comment]
Exactyly which field is wrong? They look fine. The Access time is the time of last access, which would  correctly be the current time since that's when it was accessed.

**Comment by agrajag9** [comment]
Can you compare with the output from `stat`?

**Comment by jh1113** [comment]
@agrajag9 
Thank you for your answer.



The metadata in the test file is as follows.
FileName: origin.jpg
FileModifyDate: 2019:11:12 13:31:24+09:00

![image](https://user-images.githubusercontent.com/62418986/131269983-62d30c2a-380d-4ba1-a190-132e5d9b4812.png)

The exiftool results in this file also derive the same value.
![image](https://user-images.githubusercontent.com/62418986/131270087-6938c45b-9f26-4123-a31d-8e56417d28bf.png)


Current time: 2021:08:30 09:07 +9
![image](https://user-images.githubusercontent.com/62418986/131270094-0bae0066-def9-4f99-a33b-772dd0cc99ec.png)

However, extracting metadata from this file via laika.py results in the following:

![image](https://user-images.githubusercontent.com/62418986/131270226-2ca13091-c4c2-48bd-a4e6-889f3c05f39d.png)





This file has yet to be accessed or modified.
However, the MAC outputs the current time.

**Comment by agrajag9** [comment]
Aha! I see the issue now!

**TLDR: This is expected behavior.**

If you look right above the datetimes, you'll notice a key named `SourceFile` with the value `"/dev/shm/laikaboss_tmp/tmptTw72U"` (or something like that, I can't quite read it), but you submitted the file `~/DosTest/origin.jpg`. What you're seeing is that LaikaBOSS copies `~/DosTest/origin.jpg` into `/dev/shm/laikaboss_tmp/` before scanning. This is because the file may get read many times and `/dev/shm/` is actually a tmpfs in memory and by putting a copy of the file there we significantly reduce the number of filesystem calls which are notoriously expensive. So `META_EXIFTOOL` isn't actually pointed at `~/DosTest/origin.jpg` like you expect, but at a copy of it at `/dev/shm/laikaboss_tmp/tmptTw72U`, which was just created - hence the time mismatches.

And for what it's worth, `META_EXIFTOOL` doesn't actually do any EXIF parsing itself, it relies on the 3rd-party `exiftool` module:
https://github.com/lmco/laikaboss/blob/master/laikaboss/modules/meta_exiftool.py#L45-L46

**Comment by jh1113** [comment]
@agrajag9 

Oh, thank you.
I understand, then do you know what to do to get the datatime printed correctly?
